code/logistic_sgd.py

```diff
@@ -109,6 +109,9 @@ def __init__(self, input, n_in, n_out):
         # parameters of the model
         self.params = [self.W, self.b]
 
+        # keep track of model input
+        self.input = input
+
```

Comment:
Can you check if this is needed for the other models? That way, all pickled files will have the needed information.